cgo/dtrmvbench_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build go1.7
+
+package cgo
+
+import (
+	"testing"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/testblas"
+)
+
+func BenchmarkDtrmvMed(b *testing.B) {
+	n := testblas.MediumMat
+	lda := n
+	b.Run("Inc1_UP_NT_NU", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, 1, blas.Upper, blas.NoTrans, blas.NonUnit)
+	})
+	b.Run("Inc1_UP_NT_UN", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, 1, blas.Upper, blas.NoTrans, blas.Unit)
+	})
+	b.Run("Inc1_UP_TR_NU", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, 1, blas.Upper, blas.Trans, blas.NonUnit)
+	})
+	b.Run("Inc1_UP_TR_UN", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, 1, blas.Upper, blas.Trans, blas.Unit)
+	})
+	b.Run("Inc1_LO_NT_NU", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, 1, blas.Lower, blas.NoTrans, blas.NonUnit)
+	})
+	b.Run("Inc1_LO_NT_UN", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, 1, blas.Lower, blas.NoTrans, blas.Unit)
+	})
+	b.Run("Inc1_LO_TR_NU", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, 1, blas.Lower, blas.Trans, blas.NonUnit)
+	})
+	b.Run("Inc1_LO_TR_UN", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, 1, blas.Lower, blas.Trans, blas.Unit)
+	})
+
+	incX := 5
+	b.Run("IncN_UP_NT_NU", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, incX, blas.Upper, blas.NoTrans, blas.NonUnit)
+	})
+	b.Run("IncN_UP_NT_UN", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, incX, blas.Upper, blas.NoTrans, blas.Unit)
+	})
+	b.Run("IncN_UP_TR_NU", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, incX, blas.Upper, blas.Trans, blas.NonUnit)
+	})
+	b.Run("IncN_UP_TR_UN", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, incX, blas.Upper, blas.Trans, blas.Unit)
+	})
+	b.Run("IncN_LO_NT_NU", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, incX, blas.Lower, blas.NoTrans, blas.NonUnit)
+	})
+	b.Run("IncN_LO_NT_UN", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, incX, blas.Lower, blas.NoTrans, blas.Unit)
+	})
+	b.Run("IncN_LO_TR_NU", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, incX, blas.Lower, blas.Trans, blas.NonUnit)
+	})
+	b.Run("IncN_LO_TR_UN", func(b *testing.B) {
+		testblas.DtrmvBenchmark(b, Implementation{}, n, lda, incX, blas.Lower, blas.Trans, blas.Unit)
+	})
+}
+
+func BenchmarkDtrmvLarge(b *testing.B) {
+	n := testblas.LargeMat
```

Comment:
Done. 

We can probably make a better looking function for the `str` parameter, but I figure we'll cross that bridge when we know what we want that function to look like.